### DISCUSSION TITLE
Some technical process issues in scenario of scenarioLeoReducedDynamicOrbit

### DISCUSSION THREAD

**Discussion by luoppjoker** [groops-devs/groops#248] [Q&A] [answered]
Dear developers:

Thank you for providing a convenient scenario for LEO satellite dynamic orbit determination. However, I noticed that in the scenario, the variational equations and normal equations are generated using 03groopsVariational.xml, while the specific iterative solving process is implemented in 04groopsGnssProcessing.xml. Furthermore, 04groopsGnssProcessing does not regenerate new variational equations and normal equations based on the results of the previous iteration. Considering that high-precision positions have been used in the preceding steps (compared to the usual initial orbits in the range of hundreds of meters to kilometers), it is acceptable not to generate new variational equations and normal equations.

However, in practical applications for other satellites, we can only integrate using initial position, velocity, and priori atmospheric and radiation pressure coefficients with SimulateOrbit. If we continue with the aforementioned process, will it affect the accuracy of the subsequent results?

We also attempted to further improve the accuracy of orbit determination based on the above process, including the following methods: 
Setting up a new iteration, utilizing the orbit file generated by 04groopsGnssProcessing from the previous iteration (containing only position and velocity), to repeat the tasks in 03groopsVariational.xml. However, we found this approach redundant because the results from 03groopsVariational.xml seem to show little improvement in fitting accuracy regardless of how many iterations are performed.

Therefore, we would like to consult with fellow developers if there is a  better orbit determination workflow available.

With Kind Regards,
Peng

**Comment by tmayerguerr** [comment, marked as answer]
Hi Peng,

You're right in principle, you should actually iterate. In our experience, however, this is not necessary with an orbital arc length of 1 day if the approximate orbit is accurate to within a few kilometers. 

There is already an iteration and adjustment in PreprocessingVariational, so that the accuracy is usually already better than with the classic approach. For more details see the discussion #241 .

Best regards
Torsten

**Reply by luoppjoker** [reply]
Brilliant！Thank you for your answer!!
